ui/app/components/app/connected-accounts-list/connected-accounts-list.component.js

```diff
@@ -26,31 +23,38 @@ export default class ConnectedAccountsList extends PureComponent {
       name: PropTypes.string.isRequired,
       lastActive: PropTypes.number,
     })).isRequired,
-    permissions: PropTypes.arrayOf(PropTypes.shape({
-      key: PropTypes.string.isRequired,
-    })),
+    connectAccount: PropTypes.func.isRequired,
     selectedAddress: PropTypes.string.isRequired,
-    addPermittedAccount: PropTypes.func.isRequired,
-    removePermittedAccount: PropTypes.func.isRequired,
+    removePermittedAccount: PropTypes.func,
     setSelectedAddress: PropTypes.func.isRequired,
+    shouldRenderListOptions: (props, propName, componentName) => {
+      if (props[propName]) {
+        if (typeof props[propName] !== 'boolean') {
+          return new Error(
+            `${componentName}: '${propName}' must be a boolean if provided.`
+          )
+        }
+        if (!props['removePermittedAccount']) {
+          return new Error(
+            `${componentName}: '${propName}' requires 'removePermittedAccount'.`
+          )
+        }
+      }
+    },
```

Comment:
`removePermittedAccount` is used in `disconnectAccount`, which is only called if the list item options are rendered. So, if `shouldRenderListOptions === true`, `removePermittedAccount` must be provided. Otherwise, we don't care.

I'm not completely happy with this solution, but it works. I'm very open to suggestions!